### DISCUSSION TITLE
Trading a collection of assets for a single payment

### DISCUSSION THREAD

**Discussion by hanseartic** [kalepail/stellar-quest-bounties#66] [Bounty Ideas]
# Bounty idea
Let's discuss if we can define a bounty that evolves around the idea of selling collections of assets (call them badges (noooo), NFTs or just collection).

## Mechanism to offer collection of assets as a whole for sale
Given the case someone has gathered a nice collection of assets and wants to sell them all-or-none (you wouldn't do that with your precious quest-badges, would ya?), this can currently be done in an atomic transaction under the following circumstances:
* Both parties agree to ([multi][multisig]-)sign the [transaction]
* There are not more then 99 assets included in the collection-to-be-sold (one [operation] needs to cover the counter-payment)
* The assets are offered as claimable balances (otherwise the amount of possible assets would half due to necessity to establish trustlines - or even get split three ways when removing seller's trustlines in the same TX)
* **After** the deal has sealed the seller needs to remove all trustlines manually – a presigned 'remove the trustlines' transaction could be added to the seller's account to make this more easy

## Placing an offer that can be taken by an arbitrary counterpart
It is desireable that someone does not have to advertise their collection manually and than agree on a multi-sig scenario where XDRs need to be transferred between both parties to settle the deal with the (probably not so tech-savvvy) potential buyer.

## Links
- [multisig]
- [operation]
- [transaction]
- [XDR]

[multisig]: https://developers.stellar.org/docs/glossary/multisig/
[operation]: https://developers.stellar.org/docs/glossary/operations/
[transaction]: https://developers.stellar.org/docs/glossary/transactions/
[xdr]: https://developers.stellar.org/docs/glossary/xdr/

**Comment by kalepail** [comment]
👀 Love this idea. Let me give it some thought and get back to you. This is the type of bounty I'm here for. Highly educational but also quite practical. ❤️

**Reply by hanseartic** [reply]
Thank you. I already see some different ways this could be approached but I first wanted to get the discussion started.

**Reply by kalepail** [reply]
Would be cool to explore this in the Turrets ecosystem, though not required, they will need some good examples of more real world use cases and this may be a great one.
